crates/matrix-sdk-base/src/event_cache/store/traits.rs

```diff
@@ -105,20 +107,22 @@ pub trait EventCacheStore: AsyncTraitDeps {
     /// ⚠ This is meant only for super specific use cases, where there shouldn't
     /// be any live in-memory linked chunks. In general, prefer using
     /// `EventCache::clear_all_rooms()` from the common SDK crate.
+    // TODO(bnjbvr): rename to `clear_all_linked_chunks`, since it's going to also
+    // remove all the thread linked chunks later.
     async fn clear_all_rooms_chunks(&self) -> Result<(), Self::Error>;
 
     /// Given a set of event IDs, return the duplicated events along with their
     /// position if there are any.
     async fn filter_duplicated_events(
         &self,
-        room_id: &RoomId,
+        lcid: LinkedChunkId<'_>,
         events: Vec<OwnedEventId>,
     ) -> Result<Vec<(OwnedEventId, Position)>, Self::Error>;
 
-    /// Find an event by its ID.
+    /// Find an event by its ID in a linked chunk.
     async fn find_event(
         &self,
-        room_id: &RoomId,
+        lcid: LinkedChunkId<'_>,
```

Comment:
Thinking about it, it might be better keyed on the `&RoomId`, to reflect that events are saved per room (not per linked chunk).